src/platform.ts

```diff
@@ -2,13 +2,86 @@ import fs from 'fs';
 import { LogLevel, log } from './logging';
 import { PlatformInfo } from './types';
 
+let _isWindows = false;
+let _isMacintosh = false;
+let _isLinux = false;
+
+const nodeProcess = process;
+
+// Native environment
+if (typeof nodeProcess === 'object') {
+  _isWindows = nodeProcess.platform === 'win32';
+  _isMacintosh = nodeProcess.platform === 'darwin';
+  _isLinux = nodeProcess.platform === 'linux';
+}
+
+/**
+ * This interface is intentionally not identical to node.js
+ * process because it also works in sandboxed environments
+ * where the process object is implemented differently. We
+ * define the properties here that we need for `platform`
+ * to work and nothing else.
+ */
+
+export const enum TargetPlatform {
+  WIN32_X64 = 'win32-x64',
+  WIN32_ARM64 = 'win32-arm64',
+
+  LINUX_X64 = 'linux-x64',
+  LINUX_ARM64 = 'linux-arm64',
+  LINUX_ARMHF = 'linux-armhf',
+
+  ALPINE_X64 = 'alpine-x64',
+  ALPINE_ARM64 = 'alpine-arm64',
+
+  DARWIN_X64 = 'darwin-x64',
+  DARWIN_ARM64 = 'darwin-arm64',
+
+  UNIVERSAL = 'universal',
+  UNKNOWN = 'unknown',
+  UNDEFINED = 'undefined',
+}
+
+export const enum Platform {
+  Mac,
+  Linux,
+  Windows,
+}
+
+export type PlatformName = 'Windows' | 'MacOS' | 'Linux';
+
+export function PlatformToString(platform: Platform): PlatformName {
+  switch (platform) {
+    case Platform.Mac:
+      return 'MacOS';
+    case Platform.Linux:
+      return 'Linux';
+    case Platform.Windows:
+      return 'Windows';
+  }
+}
+
+let _platform: Platform = Platform.Linux;
+if (_isMacintosh) {
+  _platform = Platform.Mac;
+} else if (_isWindows) {
+  _platform = Platform.Windows;
+} else if (_isLinux) {
+  _platform = Platform.Linux;
+}
+
+export const isWindows = _isWindows;
+export const isMacintosh = _isMacintosh;
+export const isLinux = _isLinux;
+export const platform = _platform;
```

Comment:
this is probably overkill but needed better mapping for platform names to what the actual download name is

Reply:
yes, we can probably simplify this using an enum to avoid the `toString()`, and I think the `TargetPlatform` is not necessary